Sources/StreamChat/WebSocketClient/Events/MessageEvents.swift

```diff
@@ -106,21 +106,49 @@ struct MessageUpdatedEventDTO: EventWithPayload {
     }
 }
 
-public struct MessageDeletedEvent: MessageSpecificEvent {
-    public let userId: UserId
+/// Triggered when a new message is deleted.
+public struct MessageDeletedEvent: ChannelSpecificEvent {
+    /// The user who deleted the message.
+    public let user: ChatUser
```

Comment:
AFAIK `message.deleted` can have no user if it was server-side deletion. But because `userId` is currently exposed as non-optional so we have 2 options:
1. always try to parse a user and fail for server-side triggered event -> it won't be posted
2. make `user` optional and fallback to empty identifier: `var userId: UserId { user?.id ?? "" }`

The 2nd seems better to me. WDYT?

Reply:
✅ Made the `user` optional